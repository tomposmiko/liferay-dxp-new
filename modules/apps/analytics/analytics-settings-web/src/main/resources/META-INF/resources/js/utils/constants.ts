/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

const IMAGES_PATH = Liferay.ThemeDisplay.getPathThemeImages();

export const SPRITEMAP = IMAGES_PATH + '/clay/icons.svg';

export const NOT_FOUND_GIF = `${IMAGES_PATH}/states/search_state.gif`;

export const EMPTY_STATE_GIF = `${IMAGES_PATH}/states/empty_state.gif`;

export const ERROR_MESSAGE = Liferay.Language.get(
	'an-unexpected-system-error-occurred'
);
