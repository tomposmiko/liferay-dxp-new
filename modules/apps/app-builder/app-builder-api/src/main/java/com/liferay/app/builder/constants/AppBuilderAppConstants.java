/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.app.builder.constants;

import com.liferay.portal.kernel.util.StringUtil;

/**
 * @author Bruno Farache
 * @author Gabriel Albuquerque
 */
public class AppBuilderAppConstants {

	public static final int SITE_ID_ALL = -1;

	public static final int STATUS_DEPLOYED = 0;

	public static final int STATUS_UNDEPLOYED = 1;

	public enum Status {

		DEPLOYED(STATUS_DEPLOYED, "deployed"),
		UNDEPLOYED(STATUS_UNDEPLOYED, "undeployed");

		public static Status parse(int value) {
			for (Status status : values()) {
				if (status.getValue() == value) {
					return status;
				}
			}

			return null;
		}

		public static Status parse(String label) {
			for (Status status : values()) {
				if (StringUtil.equalsIgnoreCase(label, status.getLabel())) {
					return status;
				}
			}

			return null;
		}

		public String getLabel() {
			return _label;
		}

		public int getValue() {
			return _value;
		}

		private Status(int value, String label) {
			_value = value;
			_label = label;
		}

		private final String _label;
		private final int _value;

	}

}