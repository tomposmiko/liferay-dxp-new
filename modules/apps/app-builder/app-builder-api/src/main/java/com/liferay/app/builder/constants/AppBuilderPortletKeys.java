/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.app.builder.constants;

/**
 * @author Bruno Farache
 */
public class AppBuilderPortletKeys {

	public static final String APP_BUILDER =
		"com_liferay_app_builder_web_internal_portlet_AppBuilderPortlet";

	public static final String CUSTOM_OBJECTS =
		"com_liferay_app_builder_web_internal_portlet_CustomObjectsPortlet";

	public static final String PRODUCT_MENU_APP =
		"com_liferay_app_builder_web_internal_portlet_ProductMenuAppPortlet";

	public static final String STANDALONE_APP =
		"com_liferay_app_builder_web_internal_portlet_StandaloneAppPortlet";

	public static final String WIDGET_APP =
		"com_liferay_app_builder_web_internal_portlet_WidgetAppPortlet";

}