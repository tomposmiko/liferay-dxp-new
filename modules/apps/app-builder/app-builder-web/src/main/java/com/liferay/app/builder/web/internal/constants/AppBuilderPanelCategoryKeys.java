/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.app.builder.web.internal.constants;

/**
 * @author Bruno Farache
 */
public class AppBuilderPanelCategoryKeys {

	public static final String CONTROL_PANEL_APP_BUILDER =
		"control_panel.app_builder";

	public static final String CONTROL_PANEL_APP_BUILDER_APP =
		"control_panel.app_builder_app";

}