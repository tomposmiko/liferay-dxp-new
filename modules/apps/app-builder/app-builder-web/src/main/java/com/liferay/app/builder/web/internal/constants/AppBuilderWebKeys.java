/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.app.builder.web.internal.constants;

/**
 * @author Bruno Farache
 */
public class AppBuilderWebKeys {

	public static final String APP = "APP";

	public static final String APP_DEPLOYMENT_TYPE = "APP_DEPLOYMENT_TYPE";

	public static final String APP_PORTRAIT_URL = "APP_PORTRAIT_URL";

	public static final String APP_TAB = "APP_TAB";

	public static final String APP_TAB_CONTEXT = "APP_TAB_CONTEXT";

	public static final String APPS_TABS = "APPS_TABS";

	/**
	 * @deprecated As of Athanasius (7.3.x)
	 */
	@Deprecated
	public static final String DATA_LAYOUT_IDS = "DATA_LAYOUT_IDS";

	public static final String SHOW_FORM_VIEW = "SHOW_FORM_VIEW";

	public static final String SHOW_NATIVE_OBJECTS_TAB =
		"SHOW_NATIVE_OBJECTS_TAB";

	public static final String SHOW_TABLE_VIEW = "SHOW_TABLE_VIEW";

	public static final String SHOW_TRANSLATION_MANAGER =
		"SHOW_TRANSLATION_MANAGER";

}