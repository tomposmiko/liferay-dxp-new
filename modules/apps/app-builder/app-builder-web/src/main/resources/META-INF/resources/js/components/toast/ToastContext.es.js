/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayAlert from '@clayui/alert';
import React, {createContext, useState} from 'react';

const ToastContext = createContext();

const ToastContextProvider = ({autoClose = 5000, children}) => {
	const [toasts, setToasts] = useState([]);

	const addToast = toast => {
		toast = {
			...toast,
			id: Math.floor(Math.random() * 1000)
		};

		setToasts(prevToasts => prevToasts.concat(toast));
	};

	return (
		<ToastContext.Provider value={{addToast}}>
			{children}

			<ClayAlert.ToastContainer>
				{toasts.map(({id, message, ...restProps}) => (
					<ClayAlert
						autoClose={autoClose}
						key={id}
						onClose={() => {
							setToasts(prevToasts =>
								prevToasts.filter(toast => toast.id !== id)
							);
						}}
						{...restProps}
					>
						{message}
					</ClayAlert>
				))}
			</ClayAlert.ToastContainer>
		</ToastContext.Provider>
	);
};

export {ToastContext, ToastContextProvider};
