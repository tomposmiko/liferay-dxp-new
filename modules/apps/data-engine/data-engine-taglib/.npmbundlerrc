{
	"config": {
		"imports": {
			"dynamic-data-mapping-form-builder": {
				"/": ">=2.0.0"
			},
			"dynamic-data-mapping-form-field-type": {
				"/": ">=4.0.0"
			},
			"dynamic-data-mapping-form-renderer": {
				"/": ">=4.0.0"
			}
		}
	},
	"ignore": [
		"**/*.js",
		"**/__tests__/*.js",
		"!**/*.es.js",
		"!**/*.soy.js"
	]
}