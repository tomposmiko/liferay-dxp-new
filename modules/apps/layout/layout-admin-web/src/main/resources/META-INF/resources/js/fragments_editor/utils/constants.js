/**
 * Available element config keys
 */
export const ITEM_CONFIG_KEYS = {
	backgroundColor: 'backgroundColor',
	columnSpacing: 'columnSpacing',
	containerType: 'containerType',
	padding: 'padding'
};

/**
 * Available values for containerType config key
 */
export const CONTAINER_TYPES = [
	{
		containerTypeId: 'fluid',
		label: Liferay.Language.get('fluid')
	},

	{
		containerTypeId: 'fixed',
		label: Liferay.Language.get('fixed-width')
	}
];

/**
 * Fragments Editor item borders
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ITEM_BORDERS = {
	bottom: 'border-bottom',
	top: 'border-top'
};

/**
 * Fragments Editor item types
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ITEM_TYPES = {
	column: 'fragments-editor-column',
	fragment: 'fragments-editor-fragment',
	fragmentList: 'fragments-editor-fragment-list',
	section: 'fragments-editor-section'
};