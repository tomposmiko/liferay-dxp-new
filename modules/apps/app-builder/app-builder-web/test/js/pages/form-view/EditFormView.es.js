/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayModalProvider} from '@clayui/modal';
import {act, cleanup, fireEvent, render} from '@testing-library/react';
import {DataLayoutVisitor} from 'data-engine-taglib';
import React from 'react';
import {DndProvider} from 'react-dnd';
import {HTML5Backend} from 'react-dnd-html5-backend';

import {AppContextProvider} from '../../../../src/main/resources/META-INF/resources/js/AppContext.es';
import EditFormView from '../../../../src/main/resources/META-INF/resources/js/pages/form-view/EditFormView.es';
import * as toast from '../../../../src/main/resources/META-INF/resources/js/utils/toast.es';
import {FORM_VIEW} from '../../constants.es';

const {
	EDIT_FORM_VIEW_PROPS,
	FORM_VIEW_CONTEXT,
	getDataLayoutBuilderProps,
} = FORM_VIEW;

describe('EditFormView', () => {
	let dataLayoutBuilderProps;
	let dataLayoutVisitorSpy;
	let successToastSpy;

	beforeEach(() => {
		jest.useFakeTimers();

		dataLayoutBuilderProps = getDataLayoutBuilderProps();

		dataLayoutVisitorSpy = jest
			.spyOn(DataLayoutVisitor, 'isDataLayoutEmpty')
			.mockImplementation(() => false);

		successToastSpy = jest
			.spyOn(toast, 'successToast')
			.mockImplementation(() => {});

		window.Liferay = {
			...window.Liferay,
			componentReady: () => {
				return new Promise((resolve) => {
					resolve(dataLayoutBuilderProps);
				});
			},
		};
	});

	afterEach(() => {
		jest.clearAllTimers();
		jest.restoreAllMocks();
		cleanup();
	});

	afterAll(() => {
		jest.useRealTimers();
	});

	it('renders', async () => {
		const {asFragment} = render(
			<AppContextProvider>
				<div className="tools-control-group">
					<div className="control-menu-level-1-heading" />
				</div>

				<DndProvider backend={HTML5Backend}>
					<div
						id={EDIT_FORM_VIEW_PROPS.customObjectSidebarElementId}
					/>

					<EditFormView {...EDIT_FORM_VIEW_PROPS} />
				</DndProvider>
			</AppContextProvider>
		);

		await act(async () => {
			jest.runAllTimers();
		});

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders as edit-form-view and make actions', async () => {
		const {queryByPlaceholderText, queryByText} = render(
			<AppContextProvider>
				<ClayModalProvider>
					<div className="tools-control-group">
						<div className="control-menu-level-1-heading" />
					</div>

					<DndProvider backend={HTML5Backend}>
						<div
							id={
								EDIT_FORM_VIEW_PROPS.customObjectSidebarElementId
							}
						/>

						<EditFormView {...EDIT_FORM_VIEW_PROPS} />
					</DndProvider>
				</ClayModalProvider>
			</AppContextProvider>
		);

		await act(async () => {
			jest.runAllTimers();
		});

		expect(queryByText('edit-form-view')).toBeTruthy();

		const fieldDate = queryByText('Date');

		fireEvent.click(fieldDate);

		expect(
			dataLayoutBuilderProps.dispatchAction.mock.calls[1][0]
		).toStrictEqual({
			payload: {
				fieldTypeName: 'date',
			},
			type: 'ADD_CUSTOM_OBJECT_FIELD',
		});

		expect(dataLayoutBuilderProps.dispatchAction.mock.calls.length).toBe(2);

		const formName = queryByPlaceholderText('untitled-form-view');

		expect(formName.value).toBe('FormView');

		fireEvent.change(formName, {target: {value: 'My Form View'}});

		expect(
			dataLayoutBuilderProps.dispatchAction.mock.calls[2][0]
		).toStrictEqual({
			payload: {
				name: {
					en_US: 'My Form View',
				},
			},
			type: 'UPDATE_DATA_LAYOUT_NAME',
		});

		expect(dataLayoutBuilderProps.dispatchAction.mock.calls.length).toBe(3);

		expect(dataLayoutVisitorSpy.mock.calls.length).toBe(1);
	});

	it('renders as edit-form-view, simulate a field added to the layout and save it', async () => {
		const response = JSON.stringify({});

		fetch
			.mockResponseOnce(response)
			.mockResponseOnce(response)
			.mockResponseOnce(JSON.stringify({items: []}));

		const {container, queryByText} = render(
			<AppContextProvider>
				<ClayModalProvider>
					<div className="tools-control-group">
						<div className="control-menu-level-1-heading" />
					</div>

					<DndProvider backend={HTML5Backend}>
						<div
							id={
								EDIT_FORM_VIEW_PROPS.customObjectSidebarElementId
							}
						/>

						<EditFormView {...EDIT_FORM_VIEW_PROPS} />
					</DndProvider>
				</ClayModalProvider>
			</AppContextProvider>
		);

		await act(async () => {
			jest.runAllTimers();
		});

		expect(queryByText('edit-form-view')).toBeTruthy();

		const fieldTypeName = container.querySelector(
			'[data-field-type-name="Text"]'
		);

		fireEvent.dblClick(fieldTypeName);

		expect(dataLayoutBuilderProps.dispatch.mock.calls.length).toBe(1);
		expect(dataLayoutBuilderProps.dispatch.mock.calls[0][0]).toBe(
			'fieldAdded'
		);

		const saveButton = queryByText('save');

		expect(fetch.mock.calls.length).toBe(0);

		await act(async () => {
			fireEvent.click(saveButton);
		});

		expect(fetch.mock.calls.length).toBe(2);

		const [, dataLayoutParams] = fetch.mock.calls[1];

		expect(dataLayoutParams.method).toBe('POST');
		expect(dataLayoutParams.body).toEqual(
			JSON.stringify(FORM_VIEW_CONTEXT.dataLayout)
		);

		const deleteFromObject = container.querySelector(
			'[title="delete-from-object"]'
		);

		await act(async () => {
			fireEvent.click(deleteFromObject);
		});

		expect(document.querySelector('.modal')).toBeTruthy();
		expect(successToastSpy.mock.calls.length).toBe(1);
	});

	it('renders as new-form-view and make actions', async () => {
		const {
			container,
			queryAllByPlaceholderText,
			queryAllByText,
			queryByText,
		} = render(
			<AppContextProvider>
				<div className="tools-control-group">
					<div className="control-menu-level-1-heading" />
				</div>

				<div className="change-tracking-indicator" />

				<DndProvider backend={HTML5Backend}>
					<div
						id={EDIT_FORM_VIEW_PROPS.customObjectSidebarElementId}
					/>

					<EditFormView
						{...EDIT_FORM_VIEW_PROPS}
						dataLayoutId={0}
						newCustomObject={false}
					/>
				</DndProvider>
			</AppContextProvider>
		);

		await act(async () => {
			jest.runAllTimers();
		});

		expect(container.querySelector('.publications-enabled')).toBeTruthy();

		expect(queryByText('new-form-view')).toBeTruthy();

		const searchButton = container.querySelector('svg.lexicon-icon-search');

		fireEvent.click(searchButton);

		const [search] = queryAllByPlaceholderText('search...');

		expect(search.value).toBe('');

		fireEvent.change(search, {target: {value: 'Number'}});

		expect(search.value).toBe('Number');

		expect(queryAllByText('Name').length).toBe(0);
	});

	it('renders as new-form-view and delete field', async () => {
		fetch.mockResponse(
			JSON.stringify({
				items: [
					{
						dataDefinition: {},
						dataLayouts: [],
						dataListViews: [],
					},
				],
			})
		);
		const {container, queryByText} = render(
			<AppContextProvider>
				<ClayModalProvider>
					<div className="tools-control-group">
						<div className="control-menu-level-1-heading" />
					</div>

					<DndProvider backend={HTML5Backend}>
						<div
							id={
								EDIT_FORM_VIEW_PROPS.customObjectSidebarElementId
							}
						/>

						<EditFormView {...EDIT_FORM_VIEW_PROPS} />
					</DndProvider>
				</ClayModalProvider>
			</AppContextProvider>
		);

		await act(async () => {
			jest.runAllTimers();
		});

		expect(dataLayoutBuilderProps.dispatchAction.mock.calls.length).toBe(1);
		expect(dataLayoutVisitorSpy.mock.calls.length).toBe(1);

		expect(container.querySelector('.publications-enabled')).toBeFalsy();

		const [, deleteButton] = container.querySelectorAll(
			'.field-type-remove-icon button'
		);

		await act(async () => {
			fireEvent.click(deleteButton);
		});

		act(() => {
			jest.runAllTimers();
		});

		let modal = document.querySelector('.remove-object-field-panel');

		expect(modal).toBeTruthy();

		const deleteField = queryByText('delete');

		await act(async () => {
			fireEvent.click(deleteField);
		});

		expect(
			dataLayoutBuilderProps.dispatchAction.mock.calls[1][0]
		).toStrictEqual({
			payload: {
				fieldName: 'SelectFromList',
			},
			type: 'DELETE_DATA_DEFINITION_FIELD',
		});

		act(() => {
			jest.runAllTimers();
		});

		modal = document.querySelector('.remove-object-field-panel');

		expect(modal).toBeFalsy();
	});

	it('delete field from Layout', async () => {
		fetch.mockResponse(
			JSON.stringify({
				items: [
					{
						dataDefinition: {
							defaultLanguageId: 'en_US',
						},
						dataLayouts: [
							{
								name: {
									en_US: 'Test FormView',
								},
							},
						],
						dataListViews: [
							{
								name: {
									en_US: 'Test TableView',
								},
							},
						],
					},
				],
			})
		);
		const {container, queryByText} = render(
			<AppContextProvider>
				<ClayModalProvider>
					<div className="tools-control-group">
						<div className="control-menu-level-1-heading" />
					</div>

					<DndProvider backend={HTML5Backend}>
						<div
							id={
								EDIT_FORM_VIEW_PROPS.customObjectSidebarElementId
							}
						/>

						<EditFormView {...EDIT_FORM_VIEW_PROPS} />
					</DndProvider>
				</ClayModalProvider>
			</AppContextProvider>
		);

		await act(async () => {
			jest.runAllTimers();
		});

		expect(dataLayoutBuilderProps.dispatchAction.mock.calls.length).toBe(1);
		expect(dataLayoutVisitorSpy.mock.calls.length).toBe(1);

		const deleteButton = container.querySelector(
			'.field-type-remove-icon button'
		);

		await act(async () => {
			fireEvent.click(deleteButton);
		});

		act(() => {
			jest.runAllTimers();
		});

		let modal = document.querySelector('.remove-object-field-panel');

		expect(modal).toBeTruthy();

		const deleteField = queryByText('delete');

		expect(queryByText('1. Test FormView')).toBeTruthy();
		expect(queryByText('1. Test TableView')).toBeTruthy();

		await act(async () => {
			fireEvent.click(deleteField);
		});

		const [action, payload] = dataLayoutBuilderProps.dispatch.mock.calls[0];

		expect(action).toEqual('fieldDeleted');
		expect(payload).toStrictEqual({activePage: 0, fieldName: 'Text'});

		act(() => {
			jest.runAllTimers();
		});

		modal = document.querySelector('.remove-object-field-panel');

		expect(modal).toBeFalsy();
	});
});
