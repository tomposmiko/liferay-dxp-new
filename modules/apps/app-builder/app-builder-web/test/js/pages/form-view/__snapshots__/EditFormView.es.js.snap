// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFormView renders 1`] = `
<DocumentFragment>
  <div
    class="tools-control-group"
  >
    <div
      class="control-menu-level-1-heading"
    >
      edit-form-view
    </div>
  </div>
  <div
    id="customObject"
  >
    <div
      class="data-layout-builder-sidebar"
    >
      <div
        class="sidebar sidebar-light"
      >
        <form>
          <div
            class="custom-object-sidebar-header ml-4 mr-4 mt-4 pt-2 pb-2"
          >
            <div
              class="autofit-row autofit-row-center"
            >
              <div
                class="autofit-col autofit-col-expand"
              >
                <h3>
                  My Custom Object
                </h3>
              </div>
              <div
                class="autofit-col"
              >
                <button
                  class="btn btn-monospaced btn-unstyled"
                  type="button"
                >
                  <svg
                    class="lexicon-icon lexicon-icon-search"
                    role="presentation"
                  >
                    <use
                      xlink:href="#search"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="autofit-col"
              >
                <div
                  class="dropdown custom-object-dropdown"
                >
                  <button
                    class="dropdown-toggle btn btn-monospaced btn-unstyled"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-plus"
                      role="presentation"
                    >
                      <use
                        xlink:href="#plus"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="sidebar-body"
        >
          <div
            class="custom-object-field field-type autofit-row autofit-row-center"
            data-field-type-name="Text"
            draggable="true"
          >
            <div
              class="pr-2 pl-2 autofit-col"
            >
              <span
                class="sticker data-layout-builder-field-sticker sticker-light sticker-md"
              >
                <span
                  class="sticker-overlay"
                >
                  <svg
                    class="lexicon-icon lexicon-icon-undefined"
                    role="presentation"
                  >
                    <use
                      xlink:href="#undefined"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="pr-2 autofit-col autofit-col-expand"
            >
              <h4
                class="list-group-title text-truncate"
              >
                <span>
                  Name
                </span>
              </h4>
              <p
                class="list-group-subtitle text-truncate"
              >
                <small>
                  Text 
                </small>
              </p>
            </div>
            <div
              class="autofit-col pr-2"
            />
            <div
              class="pr-2 autofit-col"
            >
              <svg
                class="lexicon-icon lexicon-icon-drag"
                role="presentation"
              >
                <use
                  xlink:href="#drag"
                />
              </svg>
            </div>
            <div
              class="field-type-remove-icon"
            >
              <button
                class="btn btn-monospaced btn-outline-borderless btn-outline-secondary"
                data-tooltip-align="right"
                data-tooltip-delay="200"
                title="delete-from-object"
                type="button"
              >
                <svg
                  class="lexicon-icon lexicon-icon-times-circle"
                  role="presentation"
                >
                  <use
                    xlink:href="#times-circle"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="custom-object-field field-type autofit-row autofit-row-center"
            data-field-type-name="SelectFromList"
            draggable="true"
          >
            <div
              class="pr-2 pl-2 autofit-col"
            >
              <span
                class="sticker data-layout-builder-field-sticker sticker-light sticker-md"
              >
                <span
                  class="sticker-overlay"
                >
                  <svg
                    class="lexicon-icon lexicon-icon-undefined"
                    role="presentation"
                  >
                    <use
                      xlink:href="#undefined"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="pr-2 autofit-col autofit-col-expand"
            >
              <h4
                class="list-group-title text-truncate"
              >
                <span>
                  Options
                </span>
              </h4>
              <p
                class="list-group-subtitle text-truncate"
              >
                <small>
                  Select from List 
                </small>
              </p>
            </div>
            <div
              class="autofit-col pr-2"
            />
            <div
              class="pr-2 autofit-col"
            >
              <svg
                class="lexicon-icon lexicon-icon-drag"
                role="presentation"
              >
                <use
                  xlink:href="#drag"
                />
              </svg>
            </div>
            <div
              class="field-type-remove-icon"
            >
              <button
                class="btn btn-monospaced btn-outline-borderless btn-outline-secondary"
                data-tooltip-align="right"
                data-tooltip-delay="200"
                title="delete-from-object"
                type="button"
              >
                <svg
                  class="lexicon-icon lexicon-icon-times-circle"
                  role="presentation"
                >
                  <use
                    xlink:href="#times-circle"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="field-type-list"
          >
            <div
              class="collapsable-panel panel panel-unstyled field-type-fieldset"
              role="tablist"
            >
              <div
                class="custom-object-field field-type-header field-type autofit-row autofit-row-center"
                data-field-type-name="Field53354166"
                draggable="true"
              >
                <div
                  class="pr-2 pl-2 autofit-col"
                >
                  <span
                    class="sticker data-layout-builder-field-sticker sticker-light sticker-md"
                  >
                    <span
                      class="sticker-overlay"
                    >
                      <svg
                        class="lexicon-icon lexicon-icon-angle-right"
                        role="presentation"
                      >
                        <use
                          xlink:href="#angle-right"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <div
                  class="pr-2 autofit-col autofit-col-expand"
                >
                  <h4
                    class="list-group-title text-truncate"
                  >
                    <span>
                      Address
                    </span>
                  </h4>
                  <p
                    class="list-group-subtitle text-truncate"
                  >
                    <small>
                      fieldset - 1 fields
                    </small>
                  </p>
                </div>
                <div
                  class="autofit-col pr-2"
                />
                <div
                  class="pr-2 autofit-col"
                >
                  <svg
                    class="lexicon-icon lexicon-icon-drag"
                    role="presentation"
                  >
                    <use
                      xlink:href="#drag"
                    />
                  </svg>
                </div>
                <div
                  class="field-type-remove-icon"
                >
                  <button
                    class="btn btn-monospaced btn-outline-borderless btn-outline-secondary"
                    data-tooltip-align="right"
                    data-tooltip-delay="200"
                    title="delete-from-object"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-times-circle"
                      role="presentation"
                    >
                      <use
                        xlink:href="#times-circle"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="panel-collapse collapse"
                role="tabpanel"
              >
                <div
                  class="panel-body"
                >
                  <div
                    class="field-type-item position-relative"
                  >
                    <div
                      class="custom-object-field-children field-type autofit-row autofit-row-center"
                      data-field-type-name="Field95700329"
                      draggable="true"
                    >
                      <div
                        class="pr-2 pl-2 autofit-col"
                      >
                        <span
                          class="sticker data-layout-builder-field-sticker sticker-light sticker-md"
                        >
                          <span
                            class="sticker-overlay"
                          >
                            <svg
                              class="lexicon-icon lexicon-icon-undefined"
                              role="presentation"
                            >
                              <use
                                xlink:href="#undefined"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                      <div
                        class="pr-2 autofit-col autofit-col-expand"
                      >
                        <h4
                          class="list-group-title text-truncate"
                        >
                          <span>
                            Text Field
                          </span>
                        </h4>
                        <p
                          class="list-group-subtitle text-truncate"
                        >
                          <small>
                            Text 
                          </small>
                        </p>
                      </div>
                      <div
                        class="autofit-col pr-2"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <nav
    class="app-builder-upper-toolbar component-tbar subnav-tbar-light tbar"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <ul
        class="tbar-nav"
      >
        <li
          class="tbar-item tbar-item-expand"
        >
          <div
            class="tbar-section"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-item"
              >
                <input
                  aria-label="untitled-form-view"
                  class="form-control form-control-inline"
                  placeholder="untitled-form-view"
                  type="text"
                  value="FormView"
                />
              </div>
            </div>
          </div>
        </li>
        <li
          class="tbar-item"
        >
          <div
            class="tbar-section"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-item"
              >
                <button
                  class="ml-3 btn btn-sm btn-secondary"
                  type="button"
                >
                  cancel
                </button>
              </div>
              <div
                class="input-group-item"
              >
                <button
                  class="ml-3 btn btn-sm btn-primary"
                  type="button"
                >
                  save
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
